src/Topologies/topology2d.jl

```diff
@@ -0,0 +1,129 @@
+
+
+"""
+    Topology2D(mesh::AbstractMesh2D, elemorder=Mesh.elements(mesh))
+
+
+
+"""
+struct Topology2D{M <: Meshes.AbstractMesh{2}, EO, OI, BF} <: AbstractTopology
+    "mesh on which the topology is constructed"
+    mesh::M
+    "`elemorder[e]` should give the `e`th element "
+    elemorder::EO
```

Comment:
In a follow up PR I think it would be good to make the element ordering / order index an explicit type we can dispatch off of as part of the topology instead of two separate arguments.